assets/src/scripts/_signals.js

```diff
@@ -5,13 +5,24 @@ import { outputs } from "./_data";
 // Store the file currently visible in the preview
 const openFile = signal();
 
+// initially empty file comments
+const fileComments = signal(
+  Object.fromEntries(Object.keys(outputs).map((output) => [output, ""]))
+);
+
 // Set each output approval status to null
 const approvedFiles = signal(
   Object.fromEntries(
-    Object.entries(outputs).map(([key]) => [key, { approved: null }])
+    Object.entries(outputs).map(([output]) => [output, { approved: null }])
   )
 );
 
+const setComment = (name, comment) => {
+  const newState = { ...fileComments.value };
+  newState[name] = comment;
+  fileComments.value = newState;
+};
+
```

Comment:
```suggestion
const setComment = (name, comment) => {
  fileComments.value = { ...fileComments.value, [name]: comment };
};
```

Spreading in JS creates a copy of the data, so you can simplify this to a one-liner.

Reply:
Yeah, it was `[name]` syntax I wasn't familiar with